src/main/java/frc/robot/VisionProcessor.java

```diff
@@ -132,19 +145,34 @@ public void process(Mat sourceFrame) {
         ArrayList<RotatedRect> initial = minimumBoundingRectangle(parent.filterContoursOutput());
         ArrayList<RotatedRect> findContours = minimumBoundingRectangle(parent.findContoursOutput());
         ArrayList<RotatedRect> ok = new GibberishRectangleFilter().filter(initial);
-        ArrayList<RotatedRect> selected = new DumbAndAloneRectangleFilter().filter(ok);
-
+        numRects = ok.size();
+
+        //reporter.getTargetAlignButtonPressed() == false || prevNumRects != numRects || frames.getTicks()<10 
+        //is the target lock logic, if all of these are false than target lock is initiated
+        if(reporter.getTargetAlignButtonPressed() == false || prevNumRects != numRects || frames.getTicks()<10){
+            index = new CenterRectangleIndexFinder().getIndex(ok);
+            prevNumRects = numRects;
+            targetLock = false;
+        }
+        else{
+            targetLock =true;
+        }        
+        ArrayList<RotatedRect> selected = new FinalVisionTargetFilter().filter(ok, index);
 
         timer.start(TIMERS.OUTPUT);
         drawRectanglesOnImage(resizedImage, initial, COLORS.BLUE);
         drawRectanglesOnImage(resizedImage, findContours, COLORS.PURPLE);
         drawRectanglesOnImage(resizedImage, ok, COLORS.RED);
-        drawRectanglesOnImage(resizedImage, selected, COLORS.YELLOW);
+        if(targetLock){
```

Comment:
see below. i think you should unconditionally draw the selected rectangles using drawRectanglesOnImage, but then just add the cross-hairs or whaever if it is locked.